File: app/src/main/java/hottopic/mit/co/nz/cleaningservice/adapter/DiscountAdapter.java
package hottopic.mit.co.nz.cleaningservice.adapter;

import android.content.Context;
import android.graphics.Paint;
import android.view.ViewGroup;
import android.widget.TextView;

import hottopic.mit.co.nz.cleaningservice.R;
import hottopic.mit.co.nz.cleaningservice.entities.orders.Discount;

public class DiscountAdapter extends BaseAdapter<Discount, DiscountAdapter.DiscountViewHolder>{
    private OnItemClickListener onItemClickListener;

    public DiscountAdapter(Context context) {
        super(context);
    }

    public interface OnItemClickListener{
        void OnItemClick(Discount discount);
    }

    public void setOnItemClickListener(DiscountAdapter.OnItemClickListener listener){
        onItemClickListener = listener;
    }

    @Override
    public DiscountViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new DiscountViewHolder(context);
    }

    class DiscountViewHolder extends BaseViewHolder<Discount>{
        private TextView tv_balance, tv_was_price, tv_price;

        public DiscountViewHolder(Context context) {
            super(context, R.layout.item_discount);
            tv_price = itemView.findViewById(R.id.tv_price);
            tv_balance = itemView.findViewById(R.id.tv_balance);
            tv_was_price = itemView.findViewById(R.id.tv_was_price);

        }


        @Override
        public void onBindViewHolder(Discount model, int position) {
            tv_price.setText(model.formatPrice());
            tv_balance.setText(model.formatBalance());
            tv_was_price.setText(model.formatBalance());
            tv_was_price.getPaint().setFlags(Paint. STRIKE_THRU_TEXT_FLAG|Paint.ANTI_ALIAS_FLAG);
            itemView.setOnClickListener(view -> {
                onItemClickListener.OnItemClick(model);
            });
        }
    }
}

File: app/src/main/java/hottopic/mit/co/nz/cleaningservice/adapter/DiscountAdapter.kt
package hottopic.mit.co.nz.cleaningservice.adapter

import android.content.Context
import android.graphics.Paint
import android.view.ViewGroup
import android.widget.TextView

import hottopic.mit.co.nz.cleaningservice.R
import hottopic.mit.co.nz.cleaningservice.entities.orders.Discount

class DiscountAdapter(context: Context) : BaseAdapter<Discount, DiscountAdapter.DiscountViewHolder>(context) {
    private var onItemClickListener: OnItemClickListener? = null

    interface OnItemClickListener {
        fun onItemClick(discount: Discount)
    }

    fun setOnItemClickListener(listener: DiscountAdapter.OnItemClickListener) {
        onItemClickListener = listener
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): DiscountViewHolder {
        return DiscountViewHolder(context)
    }

    inner class DiscountViewHolder(context: Context) : BaseViewHolder<Discount>(context, R.layout.item_discount) {
        private val tvBalance: TextView = itemView.findViewById(R.id.tv_balance)
        private val tvWasPrice: TextView = itemView.findViewById(R.id.tv_was_price)
        private val tvPrice: TextView = itemView.findViewById(R.id.tv_price)


        override fun onBindViewHolder(model: Discount, position: Int) {
            tvPrice.text = model.formatPrice()
            tvBalance.text = model.formatBalance()
            tvWasPrice.text = model.formatBalance()
            tvWasPrice.paint.flags = Paint.STRIKE_THRU_TEXT_FLAG or Paint.ANTI_ALIAS_FLAG
            itemView.setOnClickListener { onItemClickListener!!.onItemClick(model) }
        }
    }
}
